sdk/servicebus/service-bus/test/stress/testRun.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -e      # bail if any commands exit with non-zero exit codes.
+
+if [ "$1" == "" ] || [ "$2" == "" ]; then
+    echo "Usage: $0 <scenario.js file (base-name from dist/)> <service-bus-package-version (ex: dev|latest|specific-version)>"
+    exit 1
+fi
+
+# don't forget to fill out the .env file (use sample.env as your base)
+source .env
+
+# figure out the proper image name, based on the git commit
+# and the state of the current clone
+SERVICEBUS_PACKAGE_VERSION=$1
+SCENARIO_JS_BASE_NAME=$2
+KUBERNETES_ENTITY="${SCENARIO_JS_BASE_NAME,,}-$SERVICEBUS_PACKAGE_VERSION"
+
+DOCKER_IMAGE_NAME="$REGISTRY_NAME/$DOCKER_REPOSITORY_NAME:$SERVICEBUS_PACKAGE_VERSION"
+K8S_POD_NAME="$KUBERNETES_ENTITY"
+K8S_SECRET_NAME="$KUBERNETES_ENTITY"
+
+echo "Will deploy"
+echo "  docker image:${DOCKER_IMAGE_NAME}"
+echo "  to pod: ${K8S_POD_NAME}"
+echo "  with secret: ${K8S_SECRET_NAME}"
+echo "  running scenario ${SCENARIO_JS_BASE_NAME}"
+echo "  Using your default context and namespace set in kubectl"
+
+createKubernetesSecret() {
+    echo "Creating Kubernetes secret"
+
+    # create/recreate our secrets.yml
+    SECRETS_YAML=$(<secrets.yml)
+
+    # all the secrets we'll replace (marked with %value% in our secrets.yaml)
+    VARIABLES_TO_INJECT="
+        K8S_SECRET_NAME
+        SERVICEBUS_CONNECTION_STRING
+        APPINSIGHTS_INSTRUMENTATIONKEY
+    "
+
+    for name in $VARIABLES_TO_INJECT; do
+        SECRETS_YAML=${SECRETS_YAML//"%$name%"/${!name}}
+    done
+
+    echo "$SECRETS_YAML" | kubectl replace --force -f -
```

Comment:
Is there a specific reason to use `replace` instead of `apply` here? I tend to prefer the declarative commands over the imperative commands, as it will also make these scripts more flexible in the future if resources need to be updated rather than created.

If you really want to fully delete and re-create, perhaps it's worth adding a random suffix to the names so each test is trackable as a single unit (I could imagine a scenario where you want multiple instances of the test running concurrently).

Reply:
I could check but with pods (not secrets though) you can't "apply" to them easily with the entire yaml, which would be the goal (ie, we're not intending to be "gentle" with the existing pod with these updates).

In general for services though, I agree.